x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -151,114 +92,85 @@ export function ServiceOverviewTransactionsTable(props: Props) {
-        return (
-          <TruncateWithTooltip
-            text={name}
-            content={
-              <TransactionDetailLink
-                serviceName={serviceName}
-                transactionName={name}
-                transactionType={type}
-                latencyAggregationType={latencyAggregationType}
-              >
-                {name}
-              </TransactionDetailLink>
-            }
-          />
-        );
-      },
-    },
-    {
-      field: 'latency',
-      name: getLatencyAggregationTypeLabel(latencyAggregationType),
-      width: px(unit * 10),
-      render: (_, { latency }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis1"
-            compact
-            series={latency.timeseries ?? undefined}
-            valueLabel={asMillisecondDuration(latency.value)}
-          />
-        );
-      },
-    },
-    {
-      field: 'throughput',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnThroughput',
-        { defaultMessage: 'Throughput' }
-      ),
-      width: px(unit * 10),
-      render: (_, { throughput }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis0"
-            compact
-            series={throughput.timeseries ?? undefined}
-            valueLabel={asTransactionRate(throughput.value)}
-          />
-        );
-      },
-    },
-    {
-      field: 'errorRate',
-      name: i18n.translate(
-        'xpack.apm.serviceOverview.transactionsTableColumnErrorRate',
-        {
-          defaultMessage: 'Error rate',
-        }
-      ),
-      width: px(unit * 8),
-      render: (_, { errorRate }) => {
-        return (
-          <SparkPlot
-            color="euiColorVis7"
-            compact
-            series={errorRate.timeseries ?? undefined}
-            valueLabel={asPercent(errorRate.value, 1)}
-          />
-        );
-      },
+  const transactionNames = JSON.stringify(
+    currentPageTransactionGroups.map(({ name }) => name).sort()
+  );
+
+  const {
+    data: transactionGroupsStatistics,
+    status: transactionGroupsStatisticsStatus,
+  } = useFetcher(
+    (callApmApi) => {
+      if (
+        !isEmpty(requestId) &&
+        currentPageTransactionGroups.length &&
+        start &&
+        end &&
+        transactionType
+      ) {
+        return callApmApi({
+          endpoint:
+            'GET /api/apm/services/{serviceName}/transactions/groups/statistics',
+          params: {
+            path: { serviceName },
+            query: {
+              start,
+              end,
+              uiFilters: JSON.stringify(uiFilters),
+              numBuckets: 20,
+              transactionType,
+              latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+              transactionNames,
+            },
+          },
+          isCachable: true,
+        }).then((result) => {
+          return { [requestId]: result };
```

Comment:
this is nice 👍

Reply:
@cauemarcondes did you try passing `preservePreviousData: false` to `useFetcher` instead of this?

Reply:
Thanks @sqren that solves the problem without the need of requestId.

Reply:
Great

Reply:
I don't think preservePreviousData: false is a 1:1 replacement for requestId (and not sure whether Søren meant it that way either).

Reply:
I thought the idea of the requestId was to invalidate the current comparison data while new data is being fetched, and this is exactly what `preservePreviousData` does, but instead of changing the requested to a new version, it cleans up the data, so the empty state shows up and is replaced by the new data once the request is done.

Reply:
FWICT we are no back to the situation where only when the transaction groups on the current page change, the second request is refetched. But it should also change when the time range changes, or when the UI filters change, or the latency aggregation type changes.

Reply:
indeed requestId is still necessary. I added it here https://github.com/elastic/kibana/pull/88946/commits/e90fa6cba98fd4254922557f6dad500fb7de4548